src/server.cpp

```diff
@@ -2257,6 +2304,17 @@ void Server::fadeSound(s32 handle, float step, float gain)
 		m_playing_sounds.erase(it);
 }
 
+void Server::createSoundPacket(NetworkPacket &pkt, s32 sound_id, const ServerPlayingSound &params, const v3f &pos, bool ephemeral)
+{
+	float gain = params.gain * params.spec.gain;
+	pkt << sound_id << params.spec.name << gain
+			<< (u8) params.type << pos << params.object
+			<< params.spec.loop << params.spec.fade << params.spec.pitch
+			<< ephemeral 
+			<< (params.spec.start_time + m_playing_sounds_time - params.start_time);
```

Comment:
1. Why is `params.start_time` needed?
2. Please put calculations into a separate variable so that we can be sure that the network serializes the correct data type.

Reply:
`params.start_time` allows the caller to define the sound start point.
So, moving this point with time makes different players who come into the hearing distance hear a similar part of the sound.

Reply:
I see. The purpose of `ServerPlayingSound::start_time` should be clarified by leaving a comment in the header file, though. Especially because there's two `start_time` variables that have almost the identical purpose, but with the important detail that one's specified by the API and the other internally for elapsed time tracking.